lib/game_loop.rb

```diff
@@ -1,50 +1,79 @@
+require_relative './game_state'
 require_relative './guess'
-require_relative './input'
-require_relative './dialog'
 
 class GameLoop
 
-	def initialize(game_state, input = Input.new)
+	def initialize(ui, gs = GameState, g = Guess)
 		@valid = true
-		@game_state = game_state
-		@dialog = Dialog.new(game_state)
-		@input = input
+		@ui = ui
+		@gs = gs
+		@g = g
 	end
 
 	def play
+		@game_state = @gs.new
+		@ui.new_game_message
+		if @game_state.saved_game_exists?
+			@game_state.restore_state if @ui.restore?
```

Comment:
Could this saved game logic be pulled out and named appropriately? There's a few blocks in here that could be extracted out into methods